### PR TITLE
Fix catch/finally executing twice with a return in try/catch and throw in finally

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
This PR modifies the error table entry generation with try/catch/finally statements, to fix the issue of the relevant catch/finally blocks being executed more than once if the try or catch blocks contain a return statement and an error is thrown in the finally block.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/4517

### CODE REVIEW

**Comment by MaryamZi** [comment]
@SupunS added with 7c834b1.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java [hunk 1]
```diff
@@ -3100,10 +3109,20 @@ private void generateFinallyInstructions(BLangStatement statement, NodeKind... e
                     return;
                 }
             }
+            if (current.getKind() == NodeKind.RETURN
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java]: I think following scenario will not work with this logic. 

```ballerina
function main(string... args) {
  try {
      if(lengthof args > 0) {
        return;
      }
      io:println("before divide by zero");
      int i = 10/0;
  } catch (error e) {
    io:println("catch");
  } finally {
    io:println("finally");
  }
}
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java [hunk 2]
```diff
@@ -3180,17 +3223,45 @@ private void generateFinallyInstructions(BLangStatement statement) {
...
+            }
+
             if (NodeKind.TRY == parent.getKind()) {
+                boolean returnInFinally = false;
+                if (hasReturn) {
+                    //if generateFinallyInstructions is called due to a return statement being present in a try,
+                    // catch block, maintain the current IP (before code generation for the finally block)
+                    // to use as the toIP of the error table entry
+                    if (!tryCatchErrorRangeToIPStack.isEmpty()) {
+                        if (tryCatchErrorRangeToIPStack.peek() != -2) {
```
Comment (hasithaa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java]: Shall we use Constants for -2 and -1

**Comment by MaryamZi** [comment]
@hasithaa fixed with 30f601c and 09298cf

**Pull request merged**